src/System.Net.Ping/tests/FunctionalTests/UnixPingUtilityTests.cs

```diff
@@ -39,6 +40,13 @@ public class UnixPingUtilityTests
 
             string pingOutput = p.StandardOutput.ReadToEnd();
             Assert.True(p.WaitForExit(TestSettings.PingTimeout), "Ping process did not exit in " + TestSettings.PingTimeout + " ms.");
+            if (p.ExitCode == 1 || p.ExitCode == 2)
+            {
+                // Workaround known OSX bug in ping6 utility.
+                Assert.Equal(utilityPath, UnixCommandLinePing.Ping6UtilityPath);
```

Comment:
does this basically mean we're not testing ping on OSX (or put another way is this a long term change)
the tests aren't doing anything unusual as far as I know - the product would throw PingException but it's still failed from the user's point of view. maybe there should be an issue to investigate more - unless it's fixed by Apple meantime.

Reply:
This codepath will not be hit if the osx bug gets fixed. We are only handling the specific bug case.